db/migrate/20181113182615_add_vms_table.rb

```diff
@@ -0,0 +1,26 @@
+class AddVmsTable < ActiveRecord::Migration[5.1]
+  def change
+    create_table :vms do |t|
+      t.references :tenant, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref
+      t.string :uuid
+      t.string :vendor
```

Comment:
isn't the source_type same thing as :vendor?

Reply:
yeah good point